wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java

```diff
@@ -0,0 +1,235 @@
+            long timeoutExpiredMs = System.currentTimeMillis() + 25000;
+            while (isAvailable) {
+                long waitMillis = timeoutExpiredMs - System.currentTimeMillis();
+                isAvailable = checkPortAvailability(Integer.parseInt(synapseServer.getUnitTestPort()));
+
+                if (waitMillis <= 0) {
+                    // timeout expired
+                    throw new IOException("Connection refused for service in port - " + synapseServer.getUnitTestPort());
+                }
+            }
+
+            getLog().info("Unit testing agent started\n");
+
+        } else {
+            getLog().info("Testing will execute with already started unit testing agent");
+        }
+
+    }
+
+    /**
+     * Stop the Unit testing agent server.
+     */
+    private void stopTestingServer() {
+
+        if (!synapseServer.getLocalServer().isEmpty()) {
+
+            try {
+                getLog().info("Stopping unit testing agent runs on port " + synapseServer.getUnitTestPort());
+
+                //check running operating system
+                if (System.getProperty("os.name").toLowerCase().contains("win")) {
+
+                    Runtime runTime = Runtime.getRuntime();
+                    Process proc =
+                            runTime.exec("cmd /c netstat -ano | findstr " + synapseServer.getUnitTestPort());
+
+                    BufferedReader stdInput = new BufferedReader(new
+                            InputStreamReader(proc.getInputStream()));
+
+                    String stream = stdInput.readLine();
+                    if (stream != null) {
+                        int index = stream.lastIndexOf(' ');
+                        String pid = stream.substring(index);
+
+                        getLog().info("Unit testing agent runs with PID of " + pid);
+                        runTime.exec("cmd /c Taskkill /PID" + pid + " /T /F");
+                    }
+
+                } else {
+                    Process pr = Runtime.getRuntime().exec("lsof -t -i:" + synapseServer.getUnitTestPort());
+                    BufferedReader read = new BufferedReader(new InputStreamReader(pr.getInputStream()));
+                    String pid = read.readLine();
+
+                    getLog().info("Unit testing agent runs with PID of " + pid);
+                    Runtime.getRuntime().exec("kill -9 " + pid);
+                }
+
+                getLog().info("Unit testing agent stopped");
+
+            } catch (Exception e) {
+                getLog().error("Error in closing the server");
+            }
+        }
+
+    }
+
+    /**
+     * Check port availability.
+     *
+     * @param port port which want to check availability
+     * @return if available true else false
+     */
+    private static boolean checkPortAvailability(int port) {
+        boolean isAvailable;
+        try (Socket socket = new Socket()) {
+            socket.connect(new InetSocketAddress("localhost", port));
```

Comment:
Don't we need to close the socket after socket connect since you are only try to check the connectivity?

Reply:
I think Java 7's try-with-resources structure automatically handles closing the resources that the try itself opens. When I add socket.close there, it indicates that redundant close.